packages/@aws-cdk/aws-kinesisfirehose-destinations/lib/redshift-table.ts

```diff
@@ -0,0 +1,177 @@
+
+  /**
+   * The role that is attached to the Redshift cluster and will have permissions to access the intermediate bucket.
+   *
+   * If a role is provided, it must be already attached to the cluster, to avoid the 10 role per cluster limit.
+   *
+   * @default - a role will be created
+   */
+  readonly bucketAccessRole?: iam.IRole;
+}
+
+/**
+ * Redshift table delivery stream destination.
+ */
+export class RedshiftTable implements firehose.IDestination {
+  /**
+   * @param table The Redshift table that data should be inserted into.
+   */
+  constructor(private readonly table: redshift.ITable, private readonly props: RedshiftDestinationProps = {}) {
+    const cluster = table.cluster;
+    if (!cluster.publiclyAccessible) {
+      throw new Error('Redshift cluster used as delivery stream destination is not publicly accessible');
+    }
+    if (!cluster.subnetGroup?.selectedSubnets?.hasPublic) {
+      throw new Error('Redshift cluster used as delivery stream destination is not located in a public subnet');
+    }
+    if (props.user) {
+      if (props.user.cluster !== table.cluster || props.user.databaseName !== table.databaseName) {
+        throw new Error('Provided Redshift user must be located in the same Redshift cluster and database as the table');
+      }
+    }
+    if (props.retryTimeout && props.retryTimeout.toSeconds() > cdk.Duration.hours(2).toSeconds()) {
+      throw new Error(`Retry timeout must be less than 7,200 seconds (2 hours), given ${props.retryTimeout.toSeconds()} seconds`);
+    }
+    if (props.compression === Compression.SNAPPY || props.compression === Compression.HADOOP_SNAPPY || props.compression === Compression.ZIP) {
+      throw new Error(`Redshift delivery stream destination does not support ${props.compression.value} compression`);
+    }
+    if (props.s3Backup?.mode === BackupMode.FAILED) {
+      throw new Error('Redshift delivery stream destination does not support ${props.s3Backup?.mode} backup mode');
+    }
+  }
+
+  public bind(scope: Construct, options: firehose.DestinationBindOptions): firehose.DestinationConfig {
+    const {
+      compression,
+      retryTimeout,
+      bufferingInterval,
+      bufferingSize,
+      copyOptions,
+    } = this.props;
+    const cluster = this.table.cluster;
+
+    const role = this.props.role ?? new iam.Role(scope, 'Redshift Destination Role', {
+      assumedBy: new iam.ServicePrincipal('firehose.amazonaws.com'),
+    });
+
+    const endpoint = cluster.clusterEndpoint;
+    const jdbcUrl = `jdbc:redshift://${endpoint.hostname}:${cdk.Token.asString(endpoint.port)}/${this.table.databaseName}`;
+    cluster.connections.allowDefaultPortFrom(options.connections, 'Allow incoming connections from Kinesis Data Firehose');
+
+    const user = this.props.user ?? new redshift.User(scope, 'Redshift User', {
+      cluster: cluster,
+      databaseName: this.table.databaseName,
+    });
+    this.table.grant(user, redshift.TableAction.INSERT);
```

Comment:
the grant created by this needs to be added as a dependable, [below](https://github.com/aws/aws-cdk/pull/15837/files#diff-7ad3065138c0bcd78070f7f716e24f9f7e66f648475716bddf7803449799de88R162)